src/kingdom/Main.java

```diff
@@ -0,0 +1,78 @@
+        animals.add(new Mammals("Zebra",1778));
+        animals.add(new Mammals("Koala",1816));
+        animals.add(new Mammals("Sloth",1804));
+        animals.add(new Mammals("Armadillo",1758));
+        animals.add(new Mammals("Raccoon",1758));
+        animals.add(new Mammals("Bigfoot",2021));
+
+        animals.add(new Birds("Pigeon",1837));
+        animals.add(new Birds("Peacock",1821));
+        animals.add(new Birds("Toucan",1758));
+        animals.add(new Birds("Parrot",1824)); 
+        animals.add(new Birds("Swan",1758));
+
+        animals.add(new Fish("Salmon",1758));
+        animals.add(new Fish("Catfish",1817));
+        animals.add(new Fish("Perch",1758));
+
+        // List all the animals in descending order by year named
+        // List all the animals alphabetically
+        // List all the animals order by how they move
+        // List only those animals the breath with lungs
+        // List only those animals that breath with lungs and were named in 1758
+        // List only those animals that lay eggs and breath with lungs
+        // List alphabetically only those animals that were named in 1758
+
+        ///simpler method for writing a lambda expression "::"=== i->blank.method(i);
+        System.out.println("List all the animals in descending order by year named\n");
+        animals.sort(Comparator.comparingInt(AbstractAnimals::getYearDiscovered).reversed());
+        System.out.println(animals.toString());
+
+        System.out.println("List all the animals alphabetically\n");
+        animals.sort((a1,a2)->a1.getName().compareTo(a2.getName()));
+        System.out.println(animals.toString());
+
+        System.out.println("List all the animals order by how they move");
+        animals.sort((a1,a2)->a1.move().compareTo(a2.move()));
+        System.out.println(animals.toString());
+
+        System.out.println("List only those animals that breath with lungs");
+        printAnimals(animals, animal->animal.breathe()=="lungs");
+
+        System.out.println("List only those animals that breath with lungs and were named in 1758");
+        printAnimals(animals, animal->animal.breathe()=="lungs" && animal.getYearDiscovered()==1758);
+
+        System.out.println("List only those animals that lay eggs and breath with lungs");
+        printAnimals(animals, animal->animal.reproduce()=="eggs" && animal.breathe()=="lungs");
+
+        System.out.println("List alphabetically only those animals that were named in 1758");
+        animals.sort((a1,a2)->a1.getName().compareTo(a2.getName()));
+        printAnimals(animals, animal-> animal.getYearDiscovered()==1758);
+
+        //stretch
+        System.out.println("For the list of animals, list alphabetically those animals that are mammals");
+        animals.sort((a1,a2)->a1.getName().compareTo(a2.getName()));
+        printAnimals(animals, animal->animal.move()=="walks"&& animal.reproduce()=="live births"&& animal.breathe()=="lungs");
+
```

Comment:
You might want to use `compareToIgnoreCase` over plain `compareTo` to reduce the likelihood of errors springing up simply due to typos.